client/scripts/views/pages/edit_new_profile/index.tsx

```diff
@@ -0,0 +1,324 @@
+/* @jsx m */
+
+import m from 'mithril';
+import app from 'state';
+import $ from 'jquery';
+
+import Sublayout from 'views/sublayout';
+
+import { NewProfile as Profile } from '../../../../scripts/models'
+import { CWButton } from '../../components/component_kit/cw_button';
+import { CWTextInput, ValidationStatus } from '../../components/component_kit/cw_text_input';
+import { Spinner } from 'construct-ui';
+import AvatarUpload, {AvatarScope}  from '../../../views/components/avatar_upload'
+
+import 'pages/edit_new_profile.scss';
+
+type EditProfileState = {
+  address: string,
+  profile: Profile,
+  profileUpdate: Object,
+  saved: boolean,
+  failed: boolean,
+  imageUploading: boolean,
+  error: EditProfileError,
+}
+
+type GetProfileResponse = {
+  profile: any
+}
+
+enum InputFormField {
+  Email,
+  ProfileName,
+  Bio,
+  ProfileImage,
+  Website,
+} 
+
+enum EditProfileError {
+  None,
+  NoProfileFound,
+  UpdateProfileFailed,
+}
+
+const NoAddressFoundError = "No address found"
+const NoProfileFoundError = "No profile found"
+
+class EditNewProfile implements m.Component<{}, EditProfileState> {
+
+  oninit(vnode) {
+    vnode.state.address = m.route.param("address")
+    vnode.state.error = EditProfileError.None
+    this.getProfile(vnode, vnode.state.address)
+
+    // If not logged in or address not owned by logged in user
+    if (!app.isLoggedIn() || 
+      !app.user.addresses.map(addressInfo => addressInfo.address).includes(vnode.state.address)) {
+      m.route.set(`/profile/${vnode.state.address}`)
+    }
+
+    vnode.state.profileUpdate = {}
+    vnode.state.saved = false
+    vnode.state.failed = false
+    vnode.state.imageUploading = false
+  }
+
+  getProfile = async (vnode, address: string) => {
```

Comment:
I would recommend marking these internal calls private (unless they're being referenced externally) and also explicitly typing vnode if possible.

Reply:
Done